From c860bda1830ef2d57b45cd05f11335b958e501ee Mon Sep 17 00:00:00 2001
From: Dominik Poggel <pog@iesy.com>
Date: Fri, 4 Nov 2022 10:24:13 +0100
Subject: [PATCH 49/49] arm64: dts: iesy: px30: limit sdmmc-speed

Limit the SDMMC-Speed to 104MHz to prevent constant retuning. PX30 has no capability for UHSI or higher.

Remove Pin-configuration for Ethernet-PHY. Config renders the interface nonfunctional.
Remove Pin-configuration for SDMMC, duplicate from px30.dtsi
---
 .../arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts | 58 ++++++-------------
 1 file changed, 17 insertions(+), 41 deletions(-)

diff --git a/arch/arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts b/arch/arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts
index adff547df16f1..3e4931e6144f5 100644
--- a/arch/arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts
+++ b/arch/arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts
@@ -861,6 +861,7 @@
 	sd-uhs-sdr25;
 	sd-uhs-sdr50;
 	sd-uhs-sdr104;
+	max-frequency = <104000000>;
 	vqmmc-supply = <&vccio_sd>;
 	/* vmmc-supply = <&vcc_sd>; */
 	status = "okay";
@@ -1089,22 +1090,22 @@
 				<2 RK_PB6 2 &pcfg_pull_up_2ma>;
 		};
 	};
-
-	gmac {
-		rmii_pins: rmii-pins {
-			rockchip,pins =
-				/* mac_txed */
-				<2 RK_PA0 RK_FUNC_2 &pcfg_pull_none_4ma>,
-				/* mac_txd0 */
-				<2 RK_PA2 RK_FUNC_2 &pcfg_pull_none_4ma>,
-				/* mac_txd1 */
-				<2 RK_PA1 RK_FUNC_2 &pcfg_pull_none_4ma>,
-				/* mac_mdio */
-				<2 RK_PA7 RK_FUNC_2 &pcfg_pull_none_4ma>,
-				/* mac_mdc */
-				<2 RK_PB1 RK_FUNC_2 &pcfg_pull_none_4ma>;
-		};
-	};
+	// This Pin configuration disables the eth0 capabilities
+	// gmac {
+	// 	rmii_pins: rmii-pins {
+	// 		rockchip,pins =
+	// 			/* mac_txed */
+	// 			<2 RK_PA0 RK_FUNC_2 &pcfg_pull_none_4ma>,
+	// 			/* mac_txd0 */
+	// 			<2 RK_PA2 RK_FUNC_2 &pcfg_pull_none_4ma>,
+	// 			/* mac_txd1 */
+	// 			<2 RK_PA1 RK_FUNC_2 &pcfg_pull_none_4ma>,
+	// 			/* mac_mdio */
+	// 			<2 RK_PA7 RK_FUNC_2 &pcfg_pull_none_4ma>,
+	// 			/* mac_mdc */
+	// 			<2 RK_PB1 RK_FUNC_2 &pcfg_pull_none_4ma>;
+	// 	};
+	// };
 
 	gpio_bb138 {
 		pinctrl_gpio_a: pinctrl-gpio-a {
@@ -1132,31 +1133,6 @@
 		};
 	};
 
-	sdmmc {
-		sdmmc_cmd: sdmmc-cmd {
-			rockchip,pins =
-				<1 RK_PD7 RK_FUNC_1 &pcfg_pull_up_8ma>;
-		};
-
-		sdmmc_clk: sdmmc-clk {
-			rockchip,pins =
-				<1 RK_PD6 RK_FUNC_1 &pcfg_pull_none_8ma>;
-		};
-
-		sdmmc_bus1: sdmmc-bus1 {
-			rockchip,pins =
-				<1 RK_PD2 RK_FUNC_1 &pcfg_pull_up_8ma>;
-			};
-
-		sdmmc_bus4: sdmmc-bus4 {
-			rockchip,pins =
-				<1 RK_PD2 RK_FUNC_1 &pcfg_pull_up_8ma>,
-				<1 RK_PD3 RK_FUNC_1 &pcfg_pull_up_8ma>,
-				<1 RK_PD4 RK_FUNC_1 &pcfg_pull_up_8ma>,
-				<1 RK_PD5 RK_FUNC_1 &pcfg_pull_up_8ma>;
-		};
-	};
-
 	pwm1 {
 		pwm1_pin: pwm1-pin {
 			rockchip,pins =
-- 
2.30.2

